hkmc2/shared/src/main/scala/hkmc2/syntax/Parser.scala

```diff
@@ -383,11 +379,19 @@ abstract class Parser(
     case (IDENT("!", _), loc) :: _ =>
       consume
       exprCont(Deref(exprImpl(Int.MaxValue)), prec, allowNewlines = true) // TODO: the prec???
-    case (IDENT(nme, sym), loc) :: _ if nme == "new" || nme == "let" || nme == "if" || nme == "case" || nme == "region" =>
-      parseRule(Keyword.all.get(nme).get.assumeRightPrec, ParseRule.prefixRules).getOrElse(Tree.Error())
     case (IDENT(nme, sym), loc) :: _ =>
-      consume
-      exprCont(Tree.Ident(nme), prec, allowNewlines = true)
+      Keyword.all.get(nme) match
+        case S(kw) =>
```

Comment:
Keywords are not expected to be handled here! This `expr` function is only for simple expressions not starting with a keyword. This `S(kw)` case should yield an error.